demo/admin/src/products/categories/ProductCategoriesPage.tsx

```diff
@@ -45,7 +65,65 @@ const ProductCategoriesPage: React.FC = () => {
                     {(selectedId) => (
                         <SaveBoundary>
                             <FormToolbar />
-                            <ProductCategoryForm id={selectedId} />
+                            <MainContent fullHeight>
+                                <RouterTabs>
+                                    <RouterTab
+                                        forceRender={true}
+                                        path=""
+                                        label={intl.formatMessage({ id: "products.editProductCategory.formTab", defaultMessage: "Product category" })}
+                                    >
+                                        <ProductCategoryForm id={selectedId} />
+                                    </RouterTab>
+                                    <RouterTab
+                                        forceRender={true}
+                                        path="/assigned-products"
+                                        label={intl.formatMessage({
+                                            id: "products.editProductCategory.assignedProducts",
+                                            defaultMessage: "Assigned Products",
+                                        })}
+                                    >
+                                        {/*<StackSwitch initialPage="table" title="TEST">*/}
+                                        {/*    <StackPage name="table">*/}
+                                        <Box sx={{ height: "100vh" }}>
+                                            <ProductsGrid filter={{ category: { equal: selectedId } }} />
+                                            {/* TODO change button behaviour and open edit-dialog, open discussion https://github.com/vivid-planet/comet/pull/2171 */}
+                                            <EditDialog
+                                                componentsProps={{
+                                                    dialog: { fullWidth: true, maxWidth: "xl" },
+                                                    dialogContent: {
+                                                        sx: { padding: 0, paddingTop: "0 !important" /* is connected to title-style */ },
+                                                    },
+                                                }}
+                                            >
+                                                <FinalForm<{ productIds: string[] }>
```

Comment:
you should not need a form in this dialog

Reply:
alright. then I'll remove this.
But I'm still missing why it's not needed. I thought there needs to be an FinalForm-Component for fields to work (checked DialogContent, it's directly from mui, so no FinalForm... I only found SaveBoundary).
Also how do I perform submit of the mutation?

Reply:
> Also how do I perform submit of the mutation?

Should be possible like that:
```
<EditDialog>
    <Savable doSave={...} hasChanges={...} doReset={...} />
    <Grid...>
</EditDialog

```

Reply:
I moved it into a separate form-component because I think the page should not do any mutations but combine different forms, tabs, grids etc.

Reply:
Moving it around doesn't make it go away :)

I don't see a reason why you should have a form here, this makes everything much more complicated.

Reply:
I tried it now. There are some issues. Do you also mean to change implementation without a Field? because now it throws an error "useForm must be used inside of a <Form> component"

Reply:
Yes, no form and no field. No final-form code anywhere.

```
<EditDialog>
    <Savable doSave={...} hasChanges={...} doReset={...} />
    <Grid...>
</EditDialog
```

Reply:
it's done. no form no field. No final-form code anywhere. :-)

Reply:
and, do you think it is better now?

Reply:
Yes definitively. I didn't thought about this option. All implementations in our projects used the final-form approach. So that's probably not the way all developers would solve this, but I like it. And if for some reason it's needed in a form (e.g. mutation also submitting relation-data) it's still possible with a simple/small SelectionGridFieldWrapper.